### PR TITLE
Update HttpRequest.cpp.mustache - use stable 4-parameter connect

### PR DESCRIPTION
**Pull request by Jazzco**
The 3-parameter connect is known to cause issues when the creating instance is deleted. In all other mustache you already use stable 4-parameter connect. In `HttpRequest.cpp.mustache` the fix was missing.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by wing328** [comment]
thanks for the PR. please follow step 3 to update the samples.

cc @ravinikam (2017/07) @stkrwork (2017/07) @etherealjoy (2018/02) @martindelille (2018/03) @muttleyxd (2019/08)

**Review by muttleyxd — APPROVED** [review]
Looks good, please do the step 3 though

**Comment by Jazzco** [comment]
I ran the parts of step 3, committed and pushed. Put I don't see a hint that these changes came in. Any hint what might went wrong?

**Comment by Jazzco** [comment]
Ah wait - I pushed to my `master` .... need to push to `patch-1`

**Comment by Jazzco** [comment]
Reviewing the changes I see a lot of path separator changes, obviously because I ran mvn and the scripts on Windows. Is that really what you want?

**Review by MartinDelille — CHANGES_REQUESTED** [review]
Yes please push only the change regarding HttpRequest please!

**Comment by wing328** [comment]
@muttleyxd @MartinDelille when you guys have time, can you please PM me via Slack? Thanks.

**Comment by Jazzco** [comment]
> Yes please push only the change regarding HttpRequest please!

As far as I see there is nothing changed but the manual change in HttpRequest, as  long as the shifted entries of `ApiException.php` and `readme.MD` are irrelevant.

From a Windows user perspective the script I was asked to follow in step 3 is unusable in this state. Maybe it should be extended to replace all `CR/LF` by single `LF`.

**Comment by wing328** [comment]
@Jazzco tests with updated samples failed: https://github.com/OpenAPITools/openapi-generator/actions/runs/9527512386/job/26264196606?pr=18939

can you please review the test failure when you've time?

**Comment by Jazzco** [comment]
> @Jazzco tests with updated samples failed: https://github.com/OpenAPITools/openapi-generator/actions/runs/9527512386/job/26264196606?pr=18939
> 
> can you please review the test failure when you've time?

@wing328 I reviewed the change and don't see what could cause the issue. As mentioned [above](https://github.com/OpenAPITools/openapi-generator/pull/18893#issuecomment-2166019047), step 3 produces many line break changes when processed on Windows, so I reverted it. Maybe i oversaw a generated change that was necessary but I guess it makes way more sense to run this script on Linux in the current state. 

Locally generating our api and patching the results with these two changes afterwards works fine.

@ravinikam (2017/07) @stkrwork (2017/07) @etherealjoy (2018/02) @martindelille (2018/03) @muttleyxd (2019/08)
Could one of you please take over this tiny change? And maybe trigger your scripting team to fix the [linefeed-issue](https://github.com/OpenAPITools/openapi-generator/pull/18893#issuecomment-2166019047)

Regards
Jazzco

**Comment by MartinDelille** [comment]
> @muttleyxd @MartinDelille when you guys have time, can you please PM me via Slack? Thanks.

I sent you an email!

**Comment by Jazzco** [comment]
I noticed that there were some other [clazy-connect-3arg-lambda] warnings, this time in api-body, and fixed them, too.

**Comment by Jazzco** [comment]
To get this right I ran step 3 from a Linux vm and pushed the generated changes now.

**Comment by Jazzco** [comment]
I tested these changes using Qt 5.15.2 on Windows. There were no compilation or link errors.

**Comment by Jazzco** [comment]
There were () missing and it obviously worked with the 3 param version but failed with the 4 param version under macOS. Hopefully it builds now.

**Review by MartinDelille — APPROVED** [review]
Thank you so much for your contribution!

**Comment by wing328** [comment]
the CI tests failed: https://github.com/OpenAPITools/openapi-generator/actions/runs/9567139672/job/26402928612?pr=18893

please take a look when you've time.

**Comment by Jazzco** [comment]
I'm out of ideas. The linker error is way to universal to track it down to a cause.
```
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Maybe if you were able to activate the `-v` option...

**Comment by wing328** [comment]
> Maybe if you were able to activate the -v option...

please update the workflow file to activate that

```
.github/workflows/samples-cpp-qt-client.yaml 
```

**Comment by MartinDelille** [comment]
I tried to reproduce the issue locally on my mac using Qt 5.15.17 without issue.

File: .github/workflows/samples-cpp-qt-client.yaml [hunk 1]
```diff
@@ -33,4 +33,4 @@ jobs:
           tools: ${{ matrix.tools }}
       - name: Build
         working-directory: "samples/client/petstore/cpp-qt"
-        run: cmake . && cmake --build .
+        run: cmake . && cmake --build --trace .
```
Comment (MartinDelille) [inline .github/workflows/samples-cpp-qt-client.yaml:36]:
```suggestion
        run: cmake . && cmake --build . --trace
```

**Review by MartinDelille — CHANGES_REQUESTED** [review]

File: .github/workflows/samples-cpp-qt-client.yaml [hunk 2]
```diff
@@ -33,4 +33,4 @@ jobs:
           tools: ${{ matrix.tools }}
       - name: Build
         working-directory: "samples/client/petstore/cpp-qt"
-        run: cmake . && cmake --build .
+        run: cmake . && cmake --build . --trace
```
Comment (MartinDelille) [inline .github/workflows/samples-cpp-qt-client.yaml:36]:
```suggestion
        run: cmake . && cmake --build . --verbose
```

File: modules/openapi-generator/src/main/resources/cpp-qt-client/HttpRequest.cpp.mustache [hunk 3]
```diff
@@ -376,12 +376,12 @@ void {{prefix}}HttpRequestWorker::execute({{prefix}}HttpRequestInput *input) {
     }
     if (reply != nullptr) {
         reply->setParent(this);
-        connect(reply, &QNetworkReply::finished, [this, reply] {
+        connect(reply, &QNetworkReply::finished, this, [this, reply] () {
```
Comment (MartinDelille) [inline modules/openapi-generator/src/main/resources/cpp-qt-client/HttpRequest.cpp.mustache:379]:
```suggestion
        connect(reply, &QNetworkReply::finished, this, [this, reply] {
```

File: modules/openapi-generator/src/main/resources/cpp-qt-client/HttpRequest.cpp.mustache [hunk 4]
```diff
@@ -376,12 +376,12 @@ void {{prefix}}HttpRequestWorker::execute({{prefix}}HttpRequestInput *input) {
...
             on_reply_finished(reply);
         });
     }
     if (timeOutTimer.interval() > 0) {
-        QObject::connect(&timeOutTimer, &QTimer::timeout, [this, reply] {
+        QObject::connect(&timeOutTimer, &QTimer::timeout, this, [this, reply] () {
```
Comment (MartinDelille) [inline modules/openapi-generator/src/main/resources/cpp-qt-client/HttpRequest.cpp.mustache:384]:
```suggestion
        QObject::connect(&timeOutTimer, &QTimer::timeout, this, [this, reply] {
```

File: modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache [hunk 5]
```diff
@@ -700,7 +700,7 @@ void {{classname}}::{{nickname}}({{#allParams}}{{#required}}const {{{dataType}}}
 
     connect(_latestWorker, &{{prefix}}HttpRequestWorker::on_execution_finished, this, &{{classname}}::{{nickname}}Callback);
     connect(this, &{{classname}}::abortRequestsSignal, _latestWorker, &QObject::deleteLater);
-    connect(_latestWorker, &QObject::destroyed, [this](){
+    connect(_latestWorker, &QObject::destroyed, this, [this](){
```
Comment (MartinDelille) [inline modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache:703]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache [hunk 6]
```diff
@@ -730,7 +730,7 @@ void {{classname}}::{{nickname}}({{#allParams}}{{#required}}const {{{dataType}}}
...
```
Comment (MartinDelille) [inline modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache:733]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache [hunk 7]
```diff
@@ -760,7 +760,7 @@ void {{classname}}::{{nickname}}({{#allParams}}{{#required}}const {{{dataType}}}
...
```
Comment (MartinDelille) [inline modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache:763]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache [hunk 8]
```diff
@@ -790,7 +790,7 @@ void {{classname}}::{{nickname}}({{#allParams}}{{#required}}const {{{dataType}}}
...
```
Comment (MartinDelille) [inline modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache:793]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: samples/client/petstore/cpp-qt/client/PFXHttpRequest.cpp [hunk 9]
```diff
@@ -383,12 +383,12 @@ void PFXHttpRequestWorker::execute(PFXHttpRequestInput *input) {
     }
     if (reply != nullptr) {
         reply->setParent(this);
-        connect(reply, &QNetworkReply::finished, [this, reply] {
+        connect(reply, &QNetworkReply::finished, this, [this, reply] () {
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXHttpRequest.cpp:386]:
```suggestion
        connect(reply, &QNetworkReply::finished, this, [this, reply] {
```

File: samples/client/petstore/cpp-qt/client/PFXHttpRequest.cpp [hunk 10]
```diff
@@ -383,12 +383,12 @@ void PFXHttpRequestWorker::execute(PFXHttpRequestInput *input) {
...
             on_reply_finished(reply);
         });
     }
     if (timeOutTimer.interval() > 0) {
-        QObject::connect(&timeOutTimer, &QTimer::timeout, [this, reply] {
+        QObject::connect(&timeOutTimer, &QTimer::timeout, this, [this, reply] () {
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXHttpRequest.cpp:391]:
```suggestion
        QObject::connect(&timeOutTimer, &QTimer::timeout, this, [this, reply] {
```

File: samples/client/petstore/cpp-qt/client/PFXPetApi.cpp [hunk 11]
```diff
@@ -277,7 +277,7 @@ void PFXPetApi::addPet(const PFXPet &pfx_pet) {
 
     connect(_latestWorker, &PFXHttpRequestWorker::on_execution_finished, this, &PFXPetApi::addPetCallback);
     connect(this, &PFXPetApi::abortRequestsSignal, _latestWorker, &QObject::deleteLater);
-    connect(_latestWorker, &QObject::destroyed, [this](){
+    connect(_latestWorker, &QObject::destroyed, this, [this](){
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:280]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: samples/client/petstore/cpp-qt/client/PFXPetApi.cpp [hunk 12]
```diff
@@ -495,7 +495,7 @@ void PFXPetApi::deletePet(const qint64 &pet_id, const ::test_namespace::Optional
 
     connect(_latestWorker, &PFXHttpRequestWorker::on_execution_finished, this, &PFXPetApi::deletePetCallback);
     connect(this, &PFXPetApi::abortRequestsSignal, _latestWorker, &QObject::deleteLater);
-    connect(_latestWorker, &QObject::destroyed, [this](){
+    connect(_latestWorker, &QObject::destroyed, this, [this](){
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:498]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: samples/client/petstore/cpp-qt/client/PFXPetApi.cpp [hunk 13]
```diff
@@ -695,7 +695,7 @@ void PFXPetApi::findPetsByStatus(const QList<QString> &status) {
 
     connect(_latestWorker, &PFXHttpRequestWorker::on_execution_finished, this, &PFXPetApi::findPetsByStatusCallback);
     connect(this, &PFXPetApi::abortRequestsSignal, _latestWorker, &QObject::deleteLater);
-    connect(_latestWorker, &QObject::destroyed, [this](){
+    connect(_latestWorker, &QObject::destroyed, this, [this](){
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:698]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: samples/client/petstore/cpp-qt/client/PFXPetApi.cpp [hunk 14]
```diff
@@ -905,7 +905,7 @@ void PFXPetApi::findPetsByTags(const QList<QString> &tags) {
 
     connect(_latestWorker, &PFXHttpRequestWorker::on_execution_finished, this, &PFXPetApi::findPetsByTagsCallback);
     connect(this, &PFXPetApi::abortRequestsSignal, _latestWorker, &QObject::deleteLater);
-    connect(_latestWorker, &QObject::destroyed, [this](){
+    connect(_latestWorker, &QObject::destroyed, this, [this](){
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:908]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: samples/client/petstore/cpp-qt/client/PFXPetApi.cpp [hunk 15]
```diff
@@ -1127,7 +1127,7 @@ void PFXPetApi::updatePet(const PFXPet &pfx_pet) {
 
     connect(_latestWorker, &PFXHttpRequestWorker::on_execution_finished, this, &PFXPetApi::updatePetCallback);
     connect(this, &PFXPetApi::abortRequestsSignal, _latestWorker, &QObject::deleteLater);
-    connect(_latestWorker, &QObject::destroyed, [this](){
+    connect(_latestWorker, &QObject::destroyed, this, [this](){
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:1130]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: samples/client/petstore/cpp-qt/client/PFXPetApi.cpp [hunk 16]
```diff
@@ -1263,7 +1263,7 @@ void PFXPetApi::updatePetWithForm(const qint64 &pet_id, const ::test_namespace::
 
     connect(_latestWorker, &PFXHttpRequestWorker::on_execution_finished, this, &PFXPetApi::updatePetWithFormCallback);
     connect(this, &PFXPetApi::abortRequestsSignal, _latestWorker, &QObject::deleteLater);
-    connect(_latestWorker, &QObject::destroyed, [this](){
+    connect(_latestWorker, &QObject::destroyed, this, [this](){
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:1266]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

File: samples/client/petstore/cpp-qt/client/PFXPetApi.cpp [hunk 17]
```diff
@@ -1399,7 +1399,7 @@ void PFXPetApi::uploadFile(const qint64 &pet_id, const ::test_namespace::Optiona
 
     connect(_latestWorker, &PFXHttpRequestWorker::on_execution_finished, this, &PFXPetApi::uploadFileCallback);
     connect(this, &PFXPetApi::abortRequestsSignal, _latestWorker, &QObject::deleteLater);
-    connect(_latestWorker, &QObject::destroyed, [this](){
+    connect(_latestWorker, &QObject::destroyed, this, [this](){
```
Comment (MartinDelille) [inline samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:1402]:
```suggestion
    connect(_latestWorker, &QObject::destroyed, this, [this] {
```

**Review by MartinDelille — CHANGES_REQUESTED** [review]
I guess the ci didn't ran properly because none of the cpp qt code changed. This minor edit should trigger it as normal (hopping I didn't failed suggesting it)

**Comment by Jazzco** [comment]
@MartinDelille The suggestions compile locally so I committed them.

**Comment by MartinDelille** [comment]
@wing328 any idea why the CI doesn't trigger properly ?

**Comment by wing328** [comment]
triggered the workflow but some tests failed. please review when you guys have time

**Comment by MartinDelille** [comment]
I create a draft PR to troubleshoot if the problem comes from the Qt installation in the CI.

**Comment by MartinDelille** [comment]
The issue is not related to this PR changed. It should be fixed after #19049 is merged.

**Comment by wing328** [comment]
@Jazzco please pull the latest master into your branch when you've time

**Comment by MartinDelille** [comment]
@wing328 not sure the CI was triggered properly. Can you restart it by hand ?

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/actions/runs/9954301257/job/27620131137?pr=18893

please update the samples one more time

**Comment by Jazzco** [comment]
updated again

**Comment by MartinDelille** [comment]
@Jazzco can you make sure you have regenerated the sample accordingly ?

**Comment by Jazzco** [comment]
> @Jazzco can you make sure you have regenerated the sample accordingly ?

done

**Comment by MartinDelille** [comment]
@wing328 It looks like the CI wasnt' triggered properly. Any idea why ?

**Comment by Jazzco** [comment]
There was a timeout warning. Maybe some resources were busy. I merged master in again to trigger a rebuild.

**Comment by Jazzco** [comment]
Looks better now

File: modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache (hunk 8 above)
Comment (wing328) [reply modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache:783]: @Jazzco has this feedback been reviewed and incorporated into your change?

Comment (Jazzco) [reply modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache]: Suggestion applied

**Comment by Jazzco** [comment]
Now it complains in "Samples up-to-date" but I did update the samples, all 3 steps.

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/actions/runs/10248864303/job/28351198527?pr=18893

please fix it when you've time

**Comment by wing328** [comment]
> Now it complains in "Samples up-to-date" but I did update the samples, all 3 steps.

did you do `clean` (e.g. mvn clean package) when rebuilding the jar?

**Comment by Jazzco** [comment]
> did you do clean (e.g. mvn clean package) when rebuilding the jar?

Yes, all 3 steps: mvn clean, generate samples, and generate docs

**Comment by Jazzco** [comment]
Also repeated the steps and git states there's nothing to commit.

**Comment by MartinDelille** [comment]
@Jazzco Since #19297 introduced a new configuration, you should probably merge master and retry the generation to fix this errors.

File: samples/client/echo_api/php-nextgen-streaming/src/ObjectSerializer.php [hunk 18]
```diff
@@ -268,6 +268,11 @@ public static function toQueryValue(
 
         $value = $flattenArray($value, $paramName);
 
+        // https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#style-values
+        if ($openApiType === 'array' && $style === 'deepObject' && $explode) {
+            return $value;
+        }
```
Comment (MartinDelille) [inline samples/client/echo_api/php-nextgen-streaming/src/ObjectSerializer.php:274]: This change looks like to be not related to the PR.

Comment (MartinDelille) [reply samples/client/echo_api/php-nextgen-streaming/src/ObjectSerializer.php:274]: An easy fix would be just to remove it. Check there is no difference with master using `git diff martin...patch-1` before committing.

**Review by MartinDelille — APPROVED** [review]

**Comment by Jazzco** [comment]
@MartinDelille Will this be merged for the 8.0.0 release? Is anything missing?

**Comment by wing328** [comment]
@Jazzco I'll take another look tomorrow

if i understand it correctly, this is not a breaking change, right?

**Comment by wing328** [comment]
@Jazzco please also PM me via Slack when you've time: https://join.slack.com/t/openapi-generator/shared_invite/zt-12jxxd7p2-XUeQM~4pzsU9x~eGLQqX2g

thank you

**Comment by wing328** [comment]
just reviewed again and looks good

thanks again for the PR

**Pull request merged**